pallets/ddc-verification/src/lib.rs

```diff
@@ -1729,10 +1720,113 @@ pub mod pallet {
 				data_grouped.push((key, chunk.collect()));
 			}
 
-			Ok(Self::fetch_valid_aggregates_passed_challenges(data_grouped))
+			Ok(Self::fetch_valid_bucket_aggregates_passed_challenges(data_grouped))
 		}
 
-		pub(crate) fn fetch_valid_aggregates_passed_challenges(
+		pub(crate) fn challenge_and_find_valid_node_sub_aggregates_not_in_consensus(
+			cluster_id: &ClusterId,
+			era_id: DdcEra,
+			dac_nodes: &[(NodePubKey, StorageNodeParams)],
+			node_aggregates_not_in_consensus: Vec<NodeActivity>,
+		) -> Result<Vec<NodeActivity>, Vec<OCWError>> {
+			let mut node_aggregates_passed_challenges: Vec<NodeActivity> = vec![];
+			let mut node_aggregates_not_passed_challenges: Vec<NodeActivity> = vec![];
+			let number_of_identifiers = T::MAX_MERKLE_NODE_IDENTIFIER;
+
+			log::info!(
+				"🚀 Challenge process starts when Node sub aggregates are not in consensus!"
+			);
+
+			for node_aggregate_activity in node_aggregates_not_in_consensus {
+				let merkle_node_ids = Self::find_random_merkle_node_ids(
+					number_of_identifiers.into(),
+					node_aggregate_activity.number_of_gets + node_aggregate_activity.number_of_puts,
+					node_aggregate_activity.clone().node_id,
+				);
+
+				log::info!("🚀 Merkle Node Identifiers for node_aggregate_activity: node id: {:?}  identifiers: {:?}",
+						node_aggregate_activity.clone().node_id, merkle_node_ids);
+
+				let challenge_responses = Self::fetch_challenge_responses(
+					cluster_id,
+					node_aggregate_activity.clone().node_id,
+					era_id,
+					None,
+					merkle_node_ids,
+					dac_nodes,
+				)
+				.map_err(|err| vec![err])?;
+
+				log::info!(
+					"🚀 Fetched challenge response node id: {:?}   challenge_response: {:?}",
+					node_aggregate_activity.clone().node_id,
+					challenge_responses
+				);
+
+				let resulting_hash_from_leafs_and_paths =
+					Self::find_resulting_hash_from_leafs_and_paths(
+						challenge_responses,
+						cluster_id,
+						era_id,
+						None,
+						node_aggregate_activity.clone().node_id,
+					)?;
+
+				let root_challenge_responses = Self::fetch_challenge_responses(
+					cluster_id,
+					node_aggregate_activity.clone().node_id,
+					era_id,
+					None,
+					vec![1],
```

Comment:
Have you synced with @jwoloch on that according to his [previous comment](https://github.com/Cerebellum-Network/blockchain-node/pull/434#discussion_r1778248414)? Do we need to request leaves of the whole tree here (by `vec![1]`) to calculate the root hash? 
If yes, we will likely need to use the `/traverse` endpoint for getting the root hash of an Aggregate, as rebuilding the whole tree just for inferring the root hash doesn't look promising